### PR TITLE
3rd part amend

### PR DESCRIPTION
**Pull request by mauro-fanton**
### Description of Work carried through

Brief description of what/why/how.

### CODE REVIEW

File: app/controllers/helpers/TaxReturnViewModel.scala [hunk 1]
```diff
@@ -74,7 +74,12 @@ case class TaxReturnViewModel (
   }
 
   // Show or hide edit links
```
Comment (liampan) [inline app/controllers/helpers/TaxReturnViewModel.scala]: should this comment be here or for the defs below?

File: app/navigation/ReturnsJourneyNavigator.scala [hunk 2]
```diff
@@ -158,15 +163,38 @@ class ReturnsJourneyNavigator @Inject()(
...
   private def exportedPlasticPackagingWeightRoute(answers: UserAnswers, mode: Mode): Call =
-    exportedAllPlastic(answers) match {
-      case true => routes.ReturnsCheckYourAnswersController.onPageLoad()
-      case false => routes.NonExportedHumanMedicinesPlasticPackagingController.onPageLoad(mode)
+    if(mode.equals(CheckMode)) {routes.PlasticExportedByAnotherBusinessController.onPageLoad(mode)}
+    else {
+      exportedAllPlastic(answers) match {
+        case true => routes.ReturnsCheckYourAnswersController.onPageLoad()
+        case false => routes.PlasticExportedByAnotherBusinessController.onPageLoad(mode)
+      }
```
Comment (liampan) [inline app/navigation/ReturnsJourneyNavigator.scala]: //todo in another pr if you want, but can we move these away from calling the general navigator

File: app/pages/AmendExportedByAnotherBusinessPage.scala [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+ * limitations under the License.
+ */
+
+package pages
+
+import play.api.libs.json.JsPath
+
+case object AmendExportedByAnotherBusinessPage extends QuestionPage[Long] {
+
+  override def path: JsPath = JsPath \ toString
```
Comment (liampan) [inline app/pages/AmendExportedByAnotherBusinessPage.scala:23]: should these be 
```suggestion
  override def path: JsPath = JsPath \ "amend" \ toString
```

File: app/views/returns/NonExportedHumanMedicinesPlasticPackagingView.scala.html [hunk 4]
```diff
@@ -41,7 +41,7 @@
 }
 
 @heading = @{
-    if(directlyExportedYesNoAnswer) {
+    if(directlyExportedYesNoAnswer || exportedByAnotherBusinessAnswer) {
```
Comment (liampan) [inline app/views/returns/NonExportedHumanMedicinesPlasticPackagingView.scala.html]: if we move this to controller, makes testing the view easier

File: app/views/returns/NonExportedHumanMedicinesPlasticPackagingWeightView.scala.html [hunk 5]
```diff
@@ -26,12 +26,12 @@
...
 )
 
-@(amount: Long, form: Form[Long], mode: Mode, directlyExportedYesNoAnswer: Boolean)(implicit request: Request[_], messages: Messages)
+@(amount: Long, form: Form[Long], mode: Mode, directlyExportedYesNoAnswer: Boolean, exportedByAnotherBusinessAnswer: Boolean)(implicit request: Request[_], messages: Messages)
 
 @paragraphContent = { @paragraphBody(messages("nonExportedHumanMedicinesPlasticPackagingWeight.paragraph.1")) }
 
 @heading = @{
-    if(directlyExportedYesNoAnswer) {
+    if(directlyExportedYesNoAnswer || exportedByAnotherBusinessAnswer) {
```
Comment (liampan) [inline app/views/returns/NonExportedHumanMedicinesPlasticPackagingWeightView.scala.html]: ^^

File: test/controllers/AmendExportedWeightControllerSpec.scala [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+import play.api.inject.bind
+import play.api.mvc.Call
+import play.api.test.FakeRequest
+import play.api.test.Helpers._
+import uk.gov.hmrc.http.HttpResponse
+import views.html.AmendExportedWeightView
+
+import scala.concurrent.Future
+
+// todo: fix tests
```
Comment (liampan) [inline test/controllers/AmendExportedWeightControllerSpec.scala:38]: remember me

File: test/controllers/AnotherBusinessExportWeightControllerSpec.scala [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+import play.api.inject.bind
+import play.api.mvc.Call
+import play.api.test.FakeRequest
+import play.api.test.Helpers._
+import repositories.SessionRepository
+import views.html.returns.AnotherBusinessExportWeightView
+
+import scala.concurrent.Future
+
+// todo: change this to proper unit test
```
Comment (liampan) [inline test/controllers/AnotherBusinessExportWeightControllerSpec.scala]: remember me

File: test/controllers/returns/AnotherBusinessExportWeightControllerSpec.scala [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+
+  val validAnswer = 0L
+  val totalPlastic: Long = 12
+
+  lazy val anotherBusinessExportedWeightRoute = controllers.returns.routes.AnotherBusinessExportWeightController.onPageLoad(NormalMode).url
+  val answersWithPreset: UserAnswers = emptyUserAnswers
+    .set(ManufacturedPlasticPackagingWeightPage, 7L).get
+    .set(ImportedPlasticPackagingWeightPage, 5L).get
+
+  //TODO: Fix tests
```
Comment (liampan) [inline test/controllers/returns/AnotherBusinessExportWeightControllerSpec.scala]: remember me

File: app/navigation/ReturnsJourneyNavigator.scala (hunk 2 above)
Comment (liampan) [reply app/navigation/ReturnsJourneyNavigator.scala]: happened in 447

File: app/controllers/helpers/TaxReturnViewModel.scala (hunk 1 above)
Comment (mauro-fanton) [reply app/controllers/helpers/TaxReturnViewModel.scala]: for the defs below. Thanks

File: app/views/amends/AmendExportedPlasticPackagingView.scala.html [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+            Html(messages("amendExportedPlasticPackaging.bulletPoint.1")),
+            Html(messages("amendExportedPlasticPackaging.bulletPoint.2"))
+        )
+    )
+
+    @paragraphBody(messages("amendExportedPlasticPackaging.paragraph.3"))
+
+    @linkButton(
+        messageKey = "site.button.continue",
+        linkUrl = "/amend-exported-weight"
```
Comment (liampan) [inline app/views/amends/AmendExportedPlasticPackagingView.scala.html:46]: reverse routes!

Comment (liampan) [reply app/views/amends/AmendExportedPlasticPackagingView.scala.html:46]: `@controllers.routes.AmendExportedWeightController.onPageLoad`

File: app/navigation/ReturnsJourneyNavigator.scala (hunk 2 above)
Comment (mauro-fanton) [reply app/navigation/ReturnsJourneyNavigator.scala]: yes, I did that in 447 as this was only to created the most of the skelaton

File: app/pages/AmendExportedByAnotherBusinessPage.scala (hunk 3 above)
Comment (mauro-fanton) [reply app/pages/AmendExportedByAnotherBusinessPage.scala:23]: yes it should. Thanks

File: app/views/returns/NonExportedHumanMedicinesPlasticPackagingWeightView.scala.html (hunk 5 above)
Comment (mauro-fanton) [reply app/views/returns/NonExportedHumanMedicinesPlasticPackagingWeightView.scala.html]: yep we could do that. maybe as part of another piece of work as I would like to get this in for the navigation. I will put a todo comment

File: test/controllers/AmendExportedWeightControllerSpec.scala (hunk 6 above)
Comment (mauro-fanton) [reply test/controllers/AmendExportedWeightControllerSpec.scala:38]: change comment to: // todo: replace test with proper real test, that do not start the application

File: app/views/amends/AmendExportedPlasticPackagingView.scala.html (hunk 9 above)
Comment (mauro-fanton) [reply app/views/amends/AmendExportedPlasticPackagingView.scala.html:46]: just put something there as navigation not done yet.

File: test/controllers/AnotherBusinessExportWeightControllerSpec.scala (hunk 7 above)
Comment (mauro-fanton) [reply test/controllers/AnotherBusinessExportWeightControllerSpec.scala]: this is already on workflowy

File: test/controllers/returns/AnotherBusinessExportWeightControllerSpec.scala (hunk 8 above)
Comment (mauro-fanton) [reply test/controllers/returns/AnotherBusinessExportWeightControllerSpec.scala]: already on workflowy.

**Pull request merged**